File: generator/src/test/resources/swift/expected/RequestUriParamsTest/test-basic-uri-parameter-generation.output.swift
import Sunday

public class API {

  public let requestFactory: RequestFactory
  public let defaultContentTypes: [MediaType]
  public let defaultAcceptTypes: [MediaType]

  public init(
    requestFactory: RequestFactory,
    defaultContentTypes: [MediaType] = [],
    defaultAcceptTypes: [MediaType] = [.json]
  ) {
    self.requestFactory = requestFactory
    self.defaultContentTypes = defaultContentTypes
    self.defaultAcceptTypes = defaultAcceptTypes
  }

  public func fetchTest(
    def: String,
    obj: Test,
    strReq: String,
    int: Int = 5
  ) async throws -> Test {
    return try await self.requestFactory.result(
      method: .get,
      pathTemplate: "/tests/{obj}/{str-req}/{int}/{def}",
      pathParameters: [
        "def": def,
        "obj": obj,
        "str-req": strReq,
        "int": int
      ],
      queryParameters: nil,
      body: Empty.none,
      contentTypes: nil,
      acceptTypes: self.defaultAcceptTypes,
      headers: nil
    )
  }

}

File: generator/src/test/resources/kotlin/expected/sunday/RequestUriParamsTest/test-basic-uri-parameter-generation.output.kt
package io.test.service

import io.outfoxx.sunday.MediaType
import io.outfoxx.sunday.RequestFactory
import io.outfoxx.sunday.http.Method
import io.test.Test
import kotlin.Int
import kotlin.String
import kotlin.collections.List

public class API(
  public val requestFactory: RequestFactory,
  public val defaultContentTypes: List<MediaType> = listOf(),
  public val defaultAcceptTypes: List<MediaType> = listOf(MediaType.JSON),
) {
  public suspend fun fetchTest(
    def: String,
    obj: Test,
    strReq: String,
    int: Int = 5,
  ): Test = this.requestFactory
    .result(
      method = Method.Get,
      pathTemplate = "/tests/{obj}/{str-req}/{int}/{def}",
      pathParameters = mapOf(
        "def" to def,
        "obj" to obj,
        "str-req" to strReq,
        "int" to int
      ),
      acceptTypes = this.defaultAcceptTypes
    )
}
